src/services/redisSessionStorage.ts

```diff
@@ -107,13 +119,25 @@ export default class RedisSessionStorage implements ISessionStorage {
     sessionToken: SessionToken,
     walletToken: WalletToken
   ): Promise<Either<Error, boolean>> {
-    const deleteSessionToken = new Promise<Either<Error, boolean>>(resolve => {
-      // Remove the specified key. A key is ignored if it does not exist.
-      // @see https://redis.io/commands/del
-      this.redisClient.del(
-        `${sessionKeyPrefix}${sessionToken}`,
-        (err, response) => resolve(this.integerReply(err, response))
-      );
+    const keys = await new Promise<Either<Error, ReadonlyArray<string>>>(
+      resolve => {
+        this.redisClient.keys(
```

Comment:
`keys` is `O(N)`, we should avoid it, see [the warning](https://redis.io/commands/KEYS) ?

I suggest to store the active session keys in a [Set](https://redis.io/topics/data-types#sets)

Reply:
Ok, but Sets don't support expiration on single values. So a mechanism of data refresh needs to be implemented after expiration of single session tokens, removing expired values from the Set itself.

Reply:
good point, we could do this kind of cleanup on session creation, after we add a new session to the Set we cleanup the expired sessions, what do you think?

Reply:
Ok perfect. I proceed on this way

Reply:
fixed in b90f2d6